### DISCUSSION TITLE
Error: failed to load session

### DISCUSSION THREAD

**Discussion by blendPOS** [socketio/socket.io#4440] [Q&A]
Here's the setup...

Localhost, SSL, Proxy, Express, Express Session, Connect-MongoDB-Session, React

Two apps:

1. Express App, handles proxy/API for... 
2. React App

Everything seems to be working fine. SSL is good, fetch is good, mongoDB is good, IO connection is good, works fine. 

What _doesn't_ work is reloading the session prior to event use in socket io on the host sever (app 1).

`socket.use((__, next) => {
            session.reload((err) => {
                if (err) {
                    console.error(err)
                } else {
                    next()
                }
            })
        })`

Is throwing this error: `Error: failed to load session`

The session seems to be working just fine, however, storing variables and so on. Reload keeps the same session, etc. Cookies are all good. CORS is all good.

But I keep getting the vague `Error: failed to load session` error and I'm stopped in my tracks.

The socket connection works just fine otherwise and can execute events. But it can't reload the session.

Anyone have an idea of what is going?

**Comment by femto-g** [comment]
If you look at the express-session source code for the `Session.reload()` function, you'll see that `Error: failed to load session` happens when the getter of your session store fails to retrieve a session with the `sessionID` of the current session context. Which means either two things:
1. Your session store is not working as intended
2. The current session context, which in this case would be in your socket middleware, does not have the same `sessionID` as one created by your express middleware.

If you're using one of the compatible stores provided in the documentation 1 is probably not the case. It could be that the initial request from your socket is not sending your session cookie on connection. So when it reaches the express-session middleware in the `io` "route" a new session context is created with a new `sessionID`. Then when you call `Session.reload()` it looks in your session store for a session that has this id and gives you that error because there isn't one. You can fix this by adding `withCredentials: true` to your options on `io()` on the client side.

**Comment by noroxi7** [comment]
I have the same problem with passportJS, socket io and session in express, i have a web page that use express, nodejs, mongodb, with passport as authentication, i want to integrate socket io but i can´t set the same session id of socket as session after login, before login, there is a session id in socket request but there is not saved in mongo store, after login, one session is created on database but it has not the same id of the socket session id, what can it be the problem? Im stuck here, please help!
